From 24660df1161f0f8d8224434fa2a899ba0c3490e1 Mon Sep 17 00:00:00 2001
From: explosion-mental <explosion0mental@gmail.com>
Date: Tue, 26 Oct 2021 12:25:25 -0500
Subject: [PATCH] [PATCH][centertitle-proporcionally] Center the title with
 proportion to the area that the title has, unlike the other center title
 patches that center proportion to the screen size width (`m->ww`). If the
 title name is to long (if the title area it's too small for the title string)
 it will back up to the default behaviour, which is to truncate the title.

---
 dwm.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/dwm.c b/dwm.c
index 4465af1..4a8dd4d 100644
--- a/dwm.c
+++ b/dwm.c
@@ -731,7 +731,10 @@ drawbar(Monitor *m)
 	if ((w = m->ww - sw - x) > bh) {
 		if (m->sel) {
 			drw_setscheme(drw, scheme[m == selmon ? SchemeSel : SchemeNorm]);
-			drw_text(drw, x, 0, w, bh, lrpad / 2, m->sel->name, 0);
+			if (TEXTW(m->sel->name) > w) /* if the title is bigger than the width of the title rectangle, don't center */
+				drw_text(drw, x, 0, w, bh, lrpad / 2, m->sel->name, 0);
+			else /* center window title and icon */
+				drw_text(drw, x, 0, w, bh, (w - TEXTW(m->sel->name)) / 2, m->sel->name, 0);
 			if (m->sel->isfloating)
 				drw_rect(drw, x + boxs, boxs, boxw, boxw, m->sel->isfixed, 0);
 		} else {
-- 
2.33.1

